# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/xpcom/xpcAccessibleHyperText.cpp
# Commit: 5e92b9c7d479
# Full Hash: 5e92b9c7d4794af1d7ab0fb5c278c2e681a18fc6
# Author: Eitan Isaacson <eitan@monotonous.org>
# Date: 2023-01-28 21:11:06
# Regressor Bug: 1811092
# File Overlap Count: 1
# Description:
#   Bug 1811092 - P1: Add new async add/set text selection message. r=Jamie
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D167201
# ==============================================================================

diff -r 7cebaa85f4ff -r 5e92b9c7d479 accessible/xpcom/xpcAccessibleHyperText.cpp
--- a/accessible/xpcom/xpcAccessibleHyperText.cpp	Fri Jan 27 20:03:06 2023 +0000
+++ b/accessible/xpcom/xpcAccessibleHyperText.cpp	Fri Jan 27 20:15:11 2023 +0000
@@ -342,21 +342,14 @@
 
   if (aSelectionNum < 0) return NS_ERROR_INVALID_ARG;
 
-  if (mIntl->IsLocal()) {
-    if (!IntlLocal()->SetSelectionBoundsAt(aSelectionNum, aStartOffset,
-                                           aEndOffset)) {
-      return NS_ERROR_INVALID_ARG;
-    }
-  } else {
 #if defined(XP_WIN)
+  if (mIntl->IsRemote() &&
+      !StaticPrefs::accessibility_cache_enabled_AtStartup()) {
     return NS_ERROR_NOT_IMPLEMENTED;
-#else
-    if (!mIntl->AsRemote()->SetSelectionBoundsAt(aSelectionNum, aStartOffset,
-                                                 aEndOffset)) {
-      return NS_ERROR_INVALID_ARG;
-    }
+  }
 #endif
-  }
+
+  Intl()->SetSelectionBoundsAt(aSelectionNum, aStartOffset, aEndOffset);
   return NS_OK;
 }
 
@@ -364,11 +357,14 @@
 xpcAccessibleHyperText::AddSelection(int32_t aStartOffset, int32_t aEndOffset) {
   if (!mIntl) return NS_ERROR_FAILURE;
 
-  if (mIntl->IsLocal()) {
-    IntlLocal()->AddToSelection(aStartOffset, aEndOffset);
-  } else {
-    mIntl->AsRemote()->AddToSelection(aStartOffset, aEndOffset);
+#if defined(XP_WIN)
+  if (mIntl->IsRemote() &&
+      !StaticPrefs::accessibility_cache_enabled_AtStartup()) {
+    return NS_ERROR_NOT_IMPLEMENTED;
   }
+#endif
+
+  Intl()->AddToSelection(aStartOffset, aEndOffset);
   return NS_OK;
 }
 
